src/python/WMCore/Services/WMStats/WMStatsPycurl.py

```diff
@@ -0,0 +1,97 @@
+"""
+This module provides some functions to create concurrent HTTP requests
+for CouchDB, as an alternative to the sequential modules available under:
+* WMCore/Services/WMStats/WMStatsReader.getTaskJobSummaryByRequest
+* WMCore/Services/WMStats/WMStatsReader.jobDetailByTasks
+"""
+import json
+
+from urllib.parse import urljoin, urlencode
+from Utils.CertTools import ckey, cert
+from WMCore.Services.pycurl_manager import getdata as multi_getdata
+
+
+def getTaskJobSummaryByRequestPycurl(rowsSummary, sampleSize, serviceOpts):
+    """
+    Pycurl-based implementation of WMStatsReader.getTaskJobSummaryByRequest
+    :param rowsSummary: a dictionary with rows from CouchDB
+    :param sampleSize: integer with number of documents to retrieve
+    :param serviceOpts: dictionary with CouchDB options (url, db name and couchapp name)
+    :return: a dictionary with job detail
+    """
+    paramQueries = []
+    for row in rowsSummary['rows']:
+        thisQuery = {"startkey": [], "endkey": [], "numOfError": 0}
+        # row["key"] = ['workflow', 'task', 'jobstatus', 'exitCode', 'site']
+        thisQuery["startkey"] = row["key"][:4]
+        if row["key"][4]:
+            thisQuery["startkey"].append(row["key"][4])  # site
+
+        thisQuery["endkey"] = []
```

Comment:
I suggest to add full documentation of how startkey, endkey should be constructured. For instance it is not clear for me from code reading why startkey should be endkey, and then we need to add empty dict., then it is not clear why errors matching 5 keys above, etc. Bottom line, programmatically I see no issue with code flow but I don't understand a thing about query and expected results. I suggest provide full curl example of the query and at least one row of results to make things clear.

If necessary, please provide this example as docstring of the module itself. And, it will be useful to show results with and without error such that it would be clear logic of `numberOfError` 4 lines below.

Reply:
I am documenting some of this in my upcoming commit, but this is a very long query/url and also a long response. So we might have to revisit/refactor that.

Reply:
I am marking this as resolved, but feel free to unresolve it if I missed anything.